src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -139,6 +139,14 @@ private void ParseRequestLine(TRequestHandler handler, ReadOnlySpan<byte> reques
         // Consume space
         offset++;
 
+        while ((uint)offset < (uint)requestLine.Length
```

Comment:
I guess there will be a small perf hit in HTTP/1.1 parsing from this extra check. Check benchmarks?

I wonder whether it would be faster to integrate logic to skip extra spaces to the for loops in path and query string that check for a space.